From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <tobias.fischer@qut.edu.au>
Date: Thu, 29 Oct 2026 19:14:27 +1000
Subject: [PATCH] Do not log uninitialized CUDA error strings

cuGetErrorName/cuGetErrorString leave their out-parameters untouched on
failure (e.g. CUDA_ERROR_STUB_LIBRARY without a driver); logging them
segfaulted. Backport of the first commit of cdcseacave/openMVS#1316.
---
 libs/Common/UtilCUDA.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/libs/Common/UtilCUDA.h b/libs/Common/UtilCUDA.h
index 9791972..ec0b1bc 100644
--- a/libs/Common/UtilCUDA.h
+++ b/libs/Common/UtilCUDA.h
@@ -54,10 +54,13 @@ extern Devices devices;
 inline CUresult __reportCudaError(CUresult result, LPCSTR errorMessage) {
 	if (result == CUDA_SUCCESS)
 		return CUDA_SUCCESS;
+	// the out-parameters are left untouched on failure (e.g. CUDA_ERROR_STUB_LIBRARY)
 	LPCSTR szName;
-	cuGetErrorName(result, &szName);
+	if (cuGetErrorName(result, &szName) != CUDA_SUCCESS)
+		szName = "unknown";
 	LPCSTR szError;
-	cuGetErrorString(result, &szError);
+	if (cuGetErrorString(result, &szError) != CUDA_SUCCESS)
+		szError = "unknown error";
 	#ifdef _DEBUG
 	VERBOSE("CUDA error at %s:%d: %s (%s (code %d) - %s)", __FILE__, __LINE__, errorMessage, szName, static_cast<unsigned>(result), szError);
 	#else
